modules/Sources/Features/ImportWallet/ImportWallet.swift

```diff
@@ -20,29 +20,30 @@ public struct ImportWallet: Reducer {
     
     public struct State: Equatable {
         @PresentationState public var alert: AlertState<Action.Alert>?
-        public var birthdayHeight = "".redacted
+        @BindingState public var importedSeedPhrase = ""
+        @BindingState public var birthdayHeight = ""
         public var birthdayHeightValue: RedactableBlockHeight?
-        public var importedSeedPhrase = "".redacted
         public var isValidMnemonic = false
         public var wordsCount = 0
         public var isValidNumberOfWords = false
         public var maxWordsCount = 0
         public var isValidForm: Bool {
             isValidMnemonic &&
-            (birthdayHeight.data.isEmpty ||
-            (!birthdayHeight.data.isEmpty && birthdayHeightValue != nil))
+            (birthdayHeight.isEmpty ||
```

Comment:
This validation is duplicated on a function below. Also it doesn't seem to actually validate that the data contains a valid string that can be converted to BlockHeight

Reply:
If `birthdayHeightValue` in non-nil, it means we were able to construct a `BlockHeight` from the string value, so yes, in an indirect sort of way, it is validating that, but I will say that it is not particularly clearly expressed by this implementation, so I think per your other comment, I will make the `BlockHeight` property a computed property. 

As to the duplication, this particular computed property indicates whether the entire form is valid. The other one is validation a single field and is used by the view to show an error state for that one field should it be invalid.